### DISCUSSION TITLE
How to display a Scene in a jupyter python notebook (using Makie with PyCall)

### DISCUSSION THREAD

**Discussion by ea42gh** [JuliaPlots/Makie.jl#2217] [Q&A] [answered]
Given a Makie Scene in a variable `fig`,
executing `Main.fig` in a jupyter notebook produces
a print representation
`<PyCall.jlwrap Scene (400px, 400px): ...` rather than the figure.

Any suggestion for a fix?

Example code in
https://stackoverflow.com/questions/73345571/how-to-get-julia-makie-output-in-python-jupiter-notebook

**Comment by jrzingel** [comment]
I am also having this issue. Were you able to find any solutions?

**Reply by ea42gh** [reply]
no. What I ended up doing is using the pyviz panel and holoview libraries...

**Reply by asinghvi17** [reply]
Have you tried this with the new PythonCall/JuliaCall infrastructure?

**Reply by jrzingel** [reply]
Nah, I was using PyCall based roughly on this example https://github.com/binder-examples/julia-python/blob/master/python-and-julia.ipynb and calling makie from a `%%julia` cell, but I'll try PythonCall/JuliaCall out. Thanks

**Comment by jrzingel** [comment, marked as answer]
Like @asinghvi17 mentioned, using JuliaCall fixes this issue. The following code works in an Ipython notebook

```python
from juliacall import Main as jl

# Install the Makie package. This only needs to be run once
import juliapkg
juliapkg.add("CairoMakie", uuid="13f3f980-e62b-5c42-98c6-ff1f3baf88f0")

# Import CairoMakie into the Julia session
jl.seval("using CairoMakie")

import numpy as np
x = np.linspace(0.1, 10, 30)
f = lambda x: x**2 * np.sin(x + 1/x)
y = f(x)

fig, ax, lns = jl.lines(x, y, label="Simple function")
jl.axislegend(position=jl.seval(':lt'))
fig
```
